plugins/providers/hyperv/action/delete_vm.rb

```diff
@@ -9,6 +9,14 @@ def initialize(app, env)
         def call(env)
           env[:ui].info("Deleting the machine...")
           env[:machine].provider.driver.delete_vm
+          # NOTE: We remove the data directory and recreate it
+          #       to overcome an issue seen when running within
+          #       the WSL. Hyper-V will successfully remove the
+          #       VM and the files will appear to be gone, but
+          #       on a subsequent up, they will cause collisions.
+          #       This forces them to be gone for real.
+          FileUtils.rm_rf(env[:machine].data_dir)
```

Comment:
Is there any chance this could blow away a dir that was unintended here? Are there any safe guards we could put in place to prevent deleting say `/` or something?

Reply:
At this point the `data_dir` will have already been used in multiple places, and is constructed within the machine initializer so it wouldn't end up with `/` as a value.